jgrapht-core/src/main/java/org/jgrapht/alg/interval/KorteMoehringIntervalGraphRecognizer.java

```diff
@@ -57,32 +67,70 @@ public KorteMoehringIntervalGraphRecognizer(Graph<V, E> graph)
      */
     private void testIntervalGraph()
     {
-
-        //check for chordality
-        isChordal = chorInspec.isChordal();
-        if(!isChordal) 
-        {
+        // if the graph is not chordal, then it is not interval
+        if (!chordalInspector.isChordal()) {
             isIntervalGraph = false;
             return;
         }
-        
-        // init all relevant objects
-        Map<V, Integer> vertexOrder = getVertexInOrder(chorInspec.getPerfectEliminationOrder());
+
+        // get the perfect elimination order for the vertices in the graph
+        // TODO: reverse the perfect elimination order retrieved from the chordal inspector
+        List<V> perfectEliminationOrder = chordalInspector.getPerfectEliminationOrder();
+        Map<V, Integer> vertexIndexMap = getVertexIndexMap(perfectEliminationOrder);
+
         // iterate over the perfect elimination order
-        for (V u : chorInspec.getPerfectEliminationOrder()) {
-            // calculate Adj(u) - the predecessors of u
-            Set<V> predecessors = getPredecessors(vertexOrder, u);
+        for (V u : perfectEliminationOrder) {
+
+            // get the predecessors of the vertex u
+            Set<V> predecessors = getPredecessors(vertexIndexMap, u);
 
             // special case for predecessors is empty
             if (predecessors.isEmpty()) {
                 addEmptyPredecessors(u);
                 continue;
             }
 
-            // labeling phase: 
-            // 1 if one but not all vertices in a PQNode is a predecessor
-            // 2/inf if all vertices in a PQNode is a predecessor
-            Map<MPQNode,Integer> positiveLabels = labelTree(predecessors);
+            // labeling
```

Comment:
Can we get this part out of the main loop? I want to have the main loop clean and most pseudo code as possible

Reply:
done my abstracting a new method. The complete implementation will be aligned with the original pseudo code if time permits.